### ISSUE TITLE
JS replace symbol creates invalid syntax for object string keys

### ISSUE THREAD

**Issue by elijahdorman** [microsoft/TypeScript#42512]
- VSCode Version: 1.52.1
- OS Version: macOS 10.15.7

Steps to Reproduce:

1. Create an object with a property (eg, `var foo = {blah: 123}`)
2. Create a reference to the property (eg, `console.log(foo.blah)`)
3. Right-click property and click Rename Symbol
4. Enter a new symbol name with quotes (eg, `"must-quote-this"`)
5. The references will be incorrectly replaced (eg, `console.log(foo."must-quote-this")`)

Does this issue occur when all extensions are disabled?: Yes (all external extensions)

**Comment by mjbvz** [comment]
Thanks for the investigation @yume-chan!

I've moved this to the TypeScript project as I believe this should be handled by the TypeScript server. It should be able to tell where the rename was triggered and if brackets need to be inserted at each replacement location 

Here's a simple code example of the problem:

```ts
const foo = {
  bar: 1
};

console.log(foo.bar); // rename 'bar' to `a.b`
```

**Comment by yume-chan** [comment]
Related: #30879, #38251

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
